### ISSUE TITLE
Logout does not clear cookies

### ISSUE THREAD

**Issue by joshmedeski** [nhost/hasura-backend-plus#407]
When I call the `POST /auth/logout` endpoint on my SPA, the browser does not clear the cookies and the user stays logged in (calling `GET /auth/token/refresh` returns a valid token again to the client).

The endpoint is successfully giving a 204 and the following cookies are returned:
![Screen Shot 2021-01-25 at 3 24 02 PM](https://user-images.githubusercontent.com/2036594/105768026-67810400-5f21-11eb-9bb7-61bc222e7384.png)

I tried using, `{withCredentials: true}` to fix the issue and to the header but it throws an error.

Any ideas why this would be happening?

**Comment by neoplastic** [comment]
I'm wondering if its because the `res.clearCookie` lines don't contain the path to assist with matching the cookie to clear like a path being set as `/`

 https://github.com/nhost/hasura-backend-plus/blob/d50a5a7c77892416337c2f0d46da5cf2cbe59af3/src/routes/auth/logout.ts#L19

**Comment by elitan** [comment]
> I'm wondering if its because the `res.clearCookie` lines don't contain the path to assist with matching the cookie to clear like a path being set as `/`
> 
> https://github.com/nhost/hasura-backend-plus/blob/d50a5a7c77892416337c2f0d46da5cf2cbe59af3/src/routes/auth/logout.ts#L19

I think you're right:

https://stackoverflow.com/a/47484544/885968

**Comment by joshmedeski** [comment]
Figured it out!

> I tried using, {withCredentials: true} to fix the issue and to the header but it throws an error.

I was passing `withCredentials` to the body:

```ts
  return await api.post<null>("/auth/logout", { withCredentials: true });
```

But, because this is a post request with no body, I have to pass `null` to the body and pass `withCredentials` to the request config:

```ts
  return await api.post<null>("/auth/logout", null, { withCredentials: true });
```

The `withCredentials` is required in the header to communicate with the httpOnly cookies.

The `null` body POST requests are throwing me off. Now that I figured it out I'll be more careful with other requests. Thanks for investigating for me.
